### PR TITLE
Current skills component

### LINKED ISSUE #479: Create input for adding current skills as a new UI element
### Overview

Create input for adding current skills as a new UI element (based on finalized 0.4 Style Guide).

Find mockup here => [Finalized v0.4 Style Guide (Figma), page 1](https://www.figma.com/file/9c6pxjImN8MjHAZLH7zxYm/VRMS-Mocks?node-id=23471%3A1)

el-1

### Action Items

- [x] Explore existing implementation for UI elements:
- `src/sass/elements`
- DEV UI KIT web page (run locally) http://localhost:3001/dev-ui-kit 
- DEV UI KIT component https://github.com/hackforla/VRMS/blob/development/client-mvp-04/src/utils/uiKit/uiKit.js

- [x]  Create JSX for new UI element inside `/*** INPUTS ***/` section in [`src/utils/uiKit`](https://github.com/hackforla/VRMS/blob/development/client-mvp-04/src/utils/uiKit/uiKit.js)
- [x]  Add styles for new UI element inside `src/sass/elements/_inputs.scss`
- [x] Define in the component data object that simulating JSON data coming from the API:
```
const skills = {
   currentSkills: ["JavaScript", "TypeScript", "React", "Angular", "Java", "Node.js", ...]
 }
```
- [ ]  Add validation for input data using regexp inside `src/utils/validation/validation.js`:
   - make sure input is not empty
   - inputted data should have only letters
Take a look at existing validation: https://github.com/hackforla/VRMS/blob/development/client-mvp-04/src/utils/validation/validation.js
- [x] Add functionality to the component:
     - [x] When the user is typing in the input -> show skills that match with expression. Something similar to autocomplete.
   For instance: 
   inputted data: "java"
   show: "JavaScript", "Java"
     - [x]  When the user selected skill from the list and click on '+' button => add wireframing => remove it if user clicks on 'x'. 

- [x]  Make sure a new UI element looks and works like in Style Guide

### Resources/Instructions
[Finalized v0.4 Style Guide (Figma)](https://www.figma.com/file/9c6pxjImN8MjHAZLH7zxYm/VRMS-Mocks?node-id=23471%3A1)
Dev UI Kit web page (run locally) http://localhost:3001/dev-ui-kit

### PR DESCRIPTION
**Pull request by jbubar**
Here is my best attempt to resolve #479
Let me know how I can improve this component, or add any features

### CODE REVIEW

**Comment by jbubar** [comment]
<details>
    <summary>gif of the thing</summary>

skillthing


</details>

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import React, { useState, useRef } from 'react';
+import SkillItem from './skillItem';
+import DropDown from './dropdown';
+import { ReactComponent as PlusSign } from '../../../assets/images/icons/plus-sign.svg';
+
+const InputSkills = ({ options, skills, setSkills }) => {
+  // const [skills, setSkills] = useState([]);
```
Comment (drubgrubby) [inline client-mvp-04/src/components/common/inputSkills/inputSkills.js]: Remove commented out hook...unless that's going to be put into service later.

File: client-mvp-04/src/utils/uiKit/uiKit.js [hunk 2]
```diff
@@ -257,7 +269,10 @@ const DevUiKit = () => {
         </div>
         <p className={'dev-comment'}>Rounded input with icon</p>
       </div>
-
+      
+      {/*** Skills Input ***/}
+      <InputSkills options={skills.currentSkills} skills={skillsInputed} setSkills={setSkillsInputed} />
+
```
Comment (drubgrubby) [inline client-mvp-04/src/utils/uiKit/uiKit.js]: Add descriptor: "Current skills input" (or some such)

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  };
+  const showError = (skill) => {
+    setErrors([...errors, `${skill} is not one of the options`])
+  }
+  const submitSkill = (e) => {
+    if(value === "") return
+    let skill = skillOptions.find(
+      skill => (
+        skill.toLowerCase() === value.toLowerCase())
+    );
```
Comment (drubgrubby) [inline client-mvp-04/src/components/common/inputSkills/inputSkills.js:38]: When it's matching skills to input it should match to the first letter of the skill.  It seems like it's matching if it includes the letter at all: 

Screen Shot 2021-06-22 at 2 37 32 PM

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+          value={value}
+          onChange={(e) => {
+            setValue(e.target.value);
+            if (e.target.value !== '') {
+              setDisplayStatus(true);
+            }
+            e.target.parentElement.classList.remove('required-asterisk');
+          }}
+        />
+        <PlusSign onClick={submitSkill} className="input-plus" />
```
Comment (drubgrubby) [inline client-mvp-04/src/components/common/inputSkills/inputSkills.js:88]: Is it possible to have the input add when hitting the "enter" key so you don't have to hit the plus sign?  It works fine with the plus sign, but it did take me a sec to realize that I had to hit the plus sign to add the skill.

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          id="skill-input"
+          ref={inputRef}
+          onFocus={(e) => (
+            e.target.parentElement.classList.remove('required-asterisk')
+          )}
+          onBlur={(e) => {
+            let input = e.target;
+            setTimeout(() => {
+              if(input.value === ""){
+                input.parentElement.classList.add('required-asterisk');
```
Comment (drubgrubby) [inline client-mvp-04/src/components/common/inputSkills/inputSkills.js]: It seems like the asterisk should disappear if there are any skills already added to the list.  Not a big deal, but it seems like it should only bug you that it's a required field if you haven't entered anything.

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  const inputRef = useRef(null);
+  
+  
+  const removeSkill = (index) => () => {
+    setSkillOptions([...skillOptions, skills[index]])
+    setSkills(skills.filter((s, i) => i !== index));
+  };
+  const showError = (skill) => {
+    setErrors([...errors, `${skill} is not one of the options`])
+  }
```
Comment (drubgrubby) [inline client-mvp-04/src/components/common/inputSkills/inputSkills.js]: It should remove the old error when someone does a subsequent input, whether it created another error or not. Now it seems to just stack the errors, which makes me feel like I can't do anything right.

Screen Shot 2021-06-22 at 2 49 15 PM

I'd also change the error to something like: "X is not a valid input. Please select from the dropdown list."

**Review by drubgrubby — CHANGES_REQUESTED** [review]
@jbubar -> Really nice work!  This is a really full functioned feature that seems to work really well and was fun to test.  I have, of course, a few change requests nonetheless.

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 5 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js]: Ohh! that makes sense 
and a lot simpler than what I put

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 1 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js]: Thanks! did it

File: client-mvp-04/src/utils/uiKit/uiKit.js (hunk 2 above)
Comment (jbubar) [reply client-mvp-04/src/utils/uiKit/uiKit.js]: Changed this! thanks

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 3 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js:38]: DOne

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 5 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js]: done

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 6 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js]: done

File: client-mvp-04/src/components/common/inputSkills/inputSkills.js (hunk 4 above)
Comment (jbubar) [reply client-mvp-04/src/components/common/inputSkills/inputSkills.js:88]: yes.. did it!

**Review by drubgrubby — APPROVED** [review]
@jbubar -> This looks great.  Thanks for making those changes.  Nice work!

**Pull request merged**